src/aiida/cmdline/commands/cmd_code.py

```diff
@@ -234,34 +236,35 @@ def show(code):
 
 @verdi_code.command()
 @arguments.CODE()
-@arguments.OUTPUT_FILE(type=click.Path(exists=False))
-@click.option(
-    '--sort/--no-sort',
-    is_flag=True,
-    default=True,
-    help='Sort the keys of the output YAML.',
-    show_default=True,
-)
+@arguments.OUTPUT_FILE(type=click.Path(exists=False, path_type=pathlib.Path), required=False)
+@options.OVERWRITE()
+@options.SORT()
 @with_dbenv()
-def export(code, output_file, sort):
+def export(code, output_file, overwrite, sort):
     """Export code to a yaml file."""
+
     import yaml
 
     code_data = {}
 
     for key in code.Model.model_fields.keys():
-        if key == 'computer':
-            value = getattr(code, key).label
-        else:
-            value = getattr(code, key)
+        value = getattr(code, key).label if key == 'computer' else getattr(code, key)
 
         # If the attribute is not set, for example ``with_mpi`` do not export it, because the YAML won't be valid for
         # use in ``verdi code create`` since ``None`` is not a valid value on the CLI.
         if value is not None:
             code_data[key] = str(value)
 
-    with open(output_file, 'w', encoding='utf-8') as yfhandle:
-        yaml.dump(code_data, yfhandle, sort_keys=sort)
+    try:
+        output_file = generate_validate_output_file(
+            output_file=output_file, entity_label=code.label, overwrite=overwrite, appendix=f'@{code_data["computer"]}'
+        )
+    except (FileExistsError, IsADirectoryError) as e:
```

Comment:
Normally you can always create files with same name as folders in OS. 
I would say just ignore and still make the file. 
The less unnecessarily raise, less annoyed user, no?

Reply:
Here I definitely don't agree. If I accidentally specify the filename that happens to be a directory and the command just deletes the entire directory and overwrites it with a file, I'd be pretty pissed. I think the "annoyance" of having to change the output filename or specify `--overwrite` is way less than all the accidental loss of data.

Reply:
Personally, I would find it super confusing to have both, a file, and a directory with exactly the same name, even if it is theoretically possible. In any case, it should be very much an edge case, I don't expect users to have directories that end with `.yml`. And if they do, they should be notified of that, I think :D

Reply:
@sphuber I don't think that's ever possible,
`write_text()` is never going to delete your directory and replace a with a file.

I understand @GeigerJ2 's point that having a folder with `.yml` is pretty rare anyways. 
The point I wanted to make was why to specify this rare scenario anyways. But ok.